### PR TITLE
SW-1159 fire hazard implement the actions

### PR DESCRIPTION
**Pull request by JFSolution**
- refactoring
- improve/fix/add actions

### CODE REVIEW

File: octoprint_mrbeam/static/js/app/view-models/modal/high_temperature_warning.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
+$(function () {
+    function HighTemperatureWarningModalViewModel(params) {
```
Comment (caglarkilimci) [inline octoprint_mrbeam/static/js/app/view-models/modal/high_temperature_warning.js:2]: Missing docstring.

File: octoprint_mrbeam/static/js/app/view-models/modal/high_temperature_warning.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            self.audio.pause();
+            self.audio.currentTime = 0;
+        };
+    }
+
+    // view model class, parameters for constructor, container to bind to
+    OCTOPRINT_VIEWMODELS.push([
+        HighTemperatureWarningModalViewModel,
+
+        // e.g. loginStateViewModel, settingsViewModel, ...
```
Comment (caglarkilimci) [inline octoprint_mrbeam/static/js/app/view-models/modal/high_temperature_warning.js:132]: What does this comment mean?

File: octoprint_mrbeam/static/js/app/view-models/mrb_state.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+$(function () {
+    function MrbStateViewModel(parameters) {
```
Comment (caglarkilimci) [inline octoprint_mrbeam/static/js/app/view-models/mrb_state.js:2]: Missing docstring.

File: octoprint_mrbeam/static/js/app/view-models/modal/high_temperature_warning.js (hunk 2 above)
Comment (JFSolution) [reply octoprint_mrbeam/static/js/app/view-models/modal/high_temperature_warning.js:132]: That you can specify the viewmodels here this viewmodel depends on, it is the same in all viewmodels

**Review by caglarkilimci — APPROVED** [review]

**Review by caglarkilimci — APPROVED** [review]

**Pull request merged**